[BUG] WASM crash on d1 `.results()` when type is mismatched

### Is there an existing issue for this?

- [x] I have searched the existing issues

### What version of `workers-rs` are you using?

0.8.5

### What version of `wrangler` are you using?

4.110.0

### Describe the bug

when calling `.results::<_>()` to deserialize the d1 results, `workers-rs` panics if there is a type mismatch, this happens because of the `unwrap()` in this line https://github.com/cloudflare/workers-rs/blob/5f2d6c9192377451d43910098738624474196364/worker/src/d1/mod.rs#L505

### Steps To Reproduce

```rust
use serde::{Deserialize, Serialize};
use worker::*;

#[event(fetch)]
async fn fetch(_req: HttpRequest, env: Env, _ctx: Context) -> Result<worker::Response> {
    let db = env.d1("db").expect("should have d1 binding");

    match db
        .prepare("SELECT 'hello' AS list")
        .run()
        .await?
        .results::<Table>()
    {
        Ok(result) => worker::Response::from_json(&result),
        Err(e) => {
            console_error!("error deserializing results into struct: {e:?}");
            worker::Response::error("Error", 500)
        }
    }
}

#[derive(Serialize, Deserialize)]
struct Table {
    list: Option<Vec<String>>,
}
```

## Expected 
It's expected to print "error deserializing results into struct" and return 500

## Actual

```bash
✘ [ERROR] Critical Error: Rust panic: panicked at /home/v/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/worker-0.8.5/src/d1/mod.rs:491:69:

  called `Result::unwrap()` on an `Err` value: Error(JsValue(TypeError: Reflect.get called on
  non-object
  TypeError: Reflect.get called on non-object
      at Reflect.get (<anonymous>)
      at __wbg_get_c7eb1f358a7654df
  (file:///home/v/Documents/poc-d1-crash-unwrap/build/index.js:1:11580)
      at [object Object]xbf7b)
      at [object Object]x158bd)
      at [object Object]x2605e)
      at [object Object]x20d4a)
      at [object Object]x25bb1)
      at _t (file:///home/v/Documents/poc-d1-crash-unwrap/build/index.js:1:17188)
      at i (file:///home/v/Documents/poc-d1-crash-unwrap/build/index.js:2:831)))
      at __wbg_new_b667d279fd5aa943
```

--- a/test/src/d1.rs
+++ b/test/src/d1.rs
@@ -381,3 +381,27 @@ pub async fn retrive_first_none(
 
     Response::ok("ok")
 }
+
+#[worker::send]
+pub async fn deserialize_with_type_mismatch(
+    _req: Request,
+    env: Env,
+    _data: SomeSharedData,
+) -> Result<Response> {
+    let db = env.d1("DB")?;
+
+    let error = worker::query!(&db, "SELECT 'hello' AS id")
+        .await?
+        .results::<NullablePerson>();
+
+    if let Err(error) = error {
+        assert_eq!(
+            error.to_string(),
+            "Error: invalid type: string \"hello\", expected u32"
+        );
+    } else {
+        panic!("Expected .results() to propagate its error");
+    }
+
+    Response::ok("ok")
+}

--- a/test/src/router.rs
+++ b/test/src/router.rs
@@ -198,6 +198,7 @@ macro_rules! add_routes (
     add_route!($obj, get, "/d1/retrieve_optional_none", d1::retrieve_optional_none);
     add_route!($obj, get, "/d1/retrieve_optional_some", d1::retrieve_optional_some);
     add_route!($obj, get, "/d1/retrive_first_none", d1::retrive_first_none);
+    add_route!($obj, get, "/d1/deserialize_with_type_mismatch", d1::deserialize_with_type_mismatch);
     add_route!($obj, get, "/kv/get", kv::get);
     add_route!($obj, get, "/kv/get-not-found", kv::get_not_found);
     add_route!($obj, get, "/kv/list-keys", kv::list_keys);

--- a/test/tests/d1.spec.ts
+++ b/test/tests/d1.spec.ts
@@ -143,4 +143,10 @@ describe("d1", () => {
     expect(await resp.text()).toBe("ok");
     expect(resp.status).toBe(200);
   });
+
+  test("deserialize_with_type_mismatch", async () => {
+    const resp = await mf.dispatchFetch(`${mfUrl}d1/deserialize_with_type_mismatch`);
+    expect(await resp.text()).toBe("ok");
+    expect(resp.status).toBe(200);
+  });
 });

--- a/worker/src/d1/mod.rs
+++ b/worker/src/d1/mod.rs
@@ -502,7 +502,7 @@ impl D1Result {
         if let Some(results) = self.0.results()? {
             let mut vec = Vec::with_capacity(results.length() as usize);
             for result in results.iter() {
-                let result = serde_wasm_bindgen::from_value(result).unwrap();
+                let result = serde_wasm_bindgen::from_value(result)?;
                 vec.push(result);
             }
             Ok(vec)

